host/traceh.c

```diff
@@ -338,11 +340,19 @@ void oe_log_message(bool is_enclave, oe_log_level_t level, const char* message)
     // get timestamp for log
     struct tm t;
     time_t lt = time(NULL);
-    gmtime_r(&lt, &t);
-
-    char time[20];
-    strftime(time, sizeof(time), "%Y-%m-%dT%H:%M:%S", &t);
-    long int usecs = 0;
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    lt = tv.tv_sec;
+    oe_localtime(&lt, &t);
+
+#ifdef _WIN32
+    char time[25];
+    strftime(time, sizeof(time), "%Y-%m-%dT%H:%M:%S%z", &t);
```

Comment:
Why are we using different format strings for Windows vs. Linux? It seems like Linux should also be using the lowercase `%z` format for the ISO standard offset (e.g. +100) rather than the timezone code? It also seems to account for the difference in the size of your time buffers. 

Also, it would be useful to document the time format in your design doc we produce by default; it would be helpful for developers using the new public API to understand what they get and if they want to change it in the callback they provide.

Reply:
Thanks! Now it is much better. I will add these useful info to the design doc.

Reply:
Per SIG-Arch meeting this morning, the callback should change to take &t directly and so there is no need to convert to a string here.

Reply:
@dthaler Thanks for the suggestion, yes both the public api and internal api are planned to take &t. 
For this pr, such change may introduce dependency on <time.h> which is prohibited by compiling option "-nostdinc".
For the coming pr for public api, since new separate header files are introduced, it is very simple to solve the dependency problem.
So my suggestion is can we postpone such change to the next coming pr?

Reply:
Ok to cover in next coming PR.